wrappers/services.go

```diff
@@ -1880,21 +1939,38 @@ func RestartServices(svcs []*snap.AppInfo, explicitServices []string,
 	return nil
 }
 
+// ServicesEnableState returns a map of service names from the given snap,
+// together with their enable/disable status.
+func ServicesEnableState(s *snap.Info, inter Interacter) (map[string]bool, error) {
+	sysd := systemd.New(systemd.SystemMode, inter)
+	sts, err := queryServiceStatusMany(sysd, s.Services())
+	if err != nil {
+		return nil, err
+	}
+
+	// loop over all services in the snap, storing the current enable status
+	snapSvcsState := make(map[string]bool, len(sts))
+	for _, st := range sts {
+		snapSvcsState[st.name] = st.isEnabled()
+	}
+	return snapSvcsState, nil
+}
+
 // QueryDisabledServices returns a list of all currently disabled snap services
 // in the snap.
 func QueryDisabledServices(info *snap.Info, pb progress.Meter) ([]string, error) {
```

Comment:
this function does not seem to have had unit tests, maybe it's time to add some?

Reply:
I think this is a really good point - I'll add some

Reply:
added!